sale_pwa_cache/views/sale_order_views.xml

```diff
@@ -0,0 +1,158 @@
+                                {'name': 'product_history', 'string': _('History'), 'domain': [('sale_order_partner_id_product', '=', True)], 'order': [{'name': 'categ_id'}], 'records_per_page': 25},
+                                {'name': 'product_history_shipping', 'string': _('History (Shipping)'), 'domain': [('sale_order_partner_id_product', '=', True)], 'order': [{'name': 'categ_id'}], 'records_per_page': 25}
+                            ]
+                        }"
+                    >
+                        <form>
+                            <field name="state" invisible="1" />
+                            <field name="display_type" invisible="1" />
+                            <field name="currency_id" invisible="1" />
+                            <field name="company_id" invisible="1" />
+                            <field
+                                name="discount"
+                                widget="numeric_step"
+                                options="{'max': 100, 'auto_select': 1}"
+                                invisible="1"
+                            />
+                            <field
+                                name="price_unit"
+                                widget="numeric_step"
+                                options="{'auto_select': 1}"
+                                invisible="1"
+                            />
+                            <field name="name" invisible="1" />
+                            <field name="product_id" invisible="1" />
+                            <field name="order_id" invisible="1" />
+                            <field name="uom_category_id" invisible="1" />
+                            <field
+                                name="product_uom_qty"
+                                class="mb-1"
+                                widget="numeric_step"
+                                context="{
+                                'partner_id': parent.partner_id,
+                                'quantity': product_uom_qty,
+                                'pricelist': parent.pricelist_id,
+                                'uom': product_uom,
+                                'company_id': parent.company_id,
+                            }"
+                            />
+                            <field
+                                name="product_uom"
+                                force_save="1"
+                                attrs="{
+                                'readonly': [('state', 'in', ('sale','done', 'cancel'))],
+                                'required': [('display_type', '=', False)],
+                            }"
+                                context="{'company_id': parent.company_id}"
+                                class="mb-2"
+                                options="{'no_open': True, 'no_create': True, 'no_edit': True}"
+                                domain="[['category_id', '=', uom_category_id]]"
```

Comment:
No need to force this domain, but used the standard one:

https://github.com/odoo/odoo/blob/c76454f325a52ba4bb49d9dae9f760046eb39ebd/addons/sale/models/sale.py#L1280

and add `product_uom_category_id` field.